Course/Course/Presentation/Subviews/TopHeaderView.swift

```diff
@@ -0,0 +1,173 @@
+//
+//  TopHeaderView.swift
+//  Course
+//
+//  Created by  Stepanok Ivan on 04.04.2024.
+//
+
+import SwiftUI
+import Kingfisher
+import Core
+import Theme
+
+struct TopHeaderView: View {
+    
+    @ObservedObject var viewModel: CourseContainerViewModel
+    var title: String
+    @Binding var collapsed: Bool
+    var containerWidth: CGFloat
+    var animationNamespace: Namespace.ID
+    @Binding private var isAnimatingForTap: Bool
+    @Environment(\.isHorizontal) private var isHorizontal
+    
+    let collapsedHorizontalHeight: CGFloat = 230
+    let collapsedVerticalHeight: CGFloat = 260
+    let expandedHeight: CGFloat = 300
+    
+    private struct GeometryName {
+        static let backButton = "backButton"
+        static let topTabBar = "topTabBar"
+        static let blurSecondaryBg = "blurSecondaryBg"
+        static let blurPrimaryBg = "blurPrimaryBg"
+        static let blurBg = "blurBg"
+    }
```

Comment:
Do we need a struct here, how about converting it to an enum?

Reply:
Thanks! Good idea